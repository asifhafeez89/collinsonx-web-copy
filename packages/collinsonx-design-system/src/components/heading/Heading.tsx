import React from 'react';
import styled from '../../styled';

export interface HeadingProps extends React.HTMLAttributes<HTMLElement> {
  as: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'legend';
  children: React.ReactNode;
  subtitle?: String;
  subtitleColor?: string;
  padding: number;
  margin: number;
  lineHeight?: number;
}

const HeadingContainer = styled.div`
  position: relative;
  display: block;
  color: ${(props) => props.color};
`;

const Subtitle = styled.div`
  position: absolute;
  top: 14px;
  font-size: 10px;
  color: ${(props) => props.color};
`;

export const Heading: React.FC<HeadingProps> = ({
  as,
  children,
  subtitle,
  subtitleColor,
  color,
  margin,
  padding,
  lineHeight,
}: HeadingProps) => {
  const HeadingElement = as;
  return (
    <HeadingContainer color={color ?? '#000000'}>
      {subtitle && (
        <Subtitle color={subtitleColor ?? '#000000'}>{subtitle}</Subtitle>
      )}
      <HeadingElement
        style={{
          padding: padding,
          margin: margin,
          lineHeight: lineHeight,
        }}
      >
        {children}
      </HeadingElement>
    </HeadingContainer>
  );
};

export default Heading;
