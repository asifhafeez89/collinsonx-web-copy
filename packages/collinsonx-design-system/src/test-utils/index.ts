import userEvent from '@testing-library/user-event';

export * from '@testing-library/react';
export { userEvent };
export { default as render, Provider } from './render';
