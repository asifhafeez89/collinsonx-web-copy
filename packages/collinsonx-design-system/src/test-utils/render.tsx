import { RenderOptions, render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { ReactElement, ReactNode } from 'react';

export const Provider = ({ children }: { children: ReactNode }) => (
  <MantineProvider>{children}</MantineProvider>
);

type Render = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => ReturnType<typeof render>;

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: Provider as unknown as RenderOptions['wrapper'],
    ...options,
  });

export default customRender;
