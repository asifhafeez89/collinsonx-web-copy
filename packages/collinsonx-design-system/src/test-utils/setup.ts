import userEvent from '@testing-library/user-event';
import render from './render';

export default function (jsx: Parameters<typeof render>[0]) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}
