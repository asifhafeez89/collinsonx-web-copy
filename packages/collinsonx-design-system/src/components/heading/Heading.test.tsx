import renderer from 'react-test-renderer';
import { Provider } from 'test-utils';
import { Heading } from './Heading';

describe('<Heading />', () => {
  it('renders heading', () => {
    const tree = renderer
      .create(
        <Provider>
          <Heading as="h2" className="foo" padding={0} margin={0}>
            Heading Title
          </Heading>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
