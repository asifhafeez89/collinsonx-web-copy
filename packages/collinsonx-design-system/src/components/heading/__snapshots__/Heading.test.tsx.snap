// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> renders heading 1`] = `
[
  <style
    dangerouslySetInnerHTML={
      {
        "__html": "@media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}",
      }
    }
    data-mantine-styles="classes"
  />,
  .emotion-0 {
  position: relative;
  display: block;
  color: #000000;
}

<div
    className="emotion-0"
    color="#000000"
  >
    <h2
      style={
        {
          "lineHeight": undefined,
          "margin": 0,
          "padding": 0,
        }
      }
    >
      Heading Title
    </h2>
  </div>,
]
`;
